File: bench/serialization/1.go
package main

import (
	"bytes"
	"encoding/gob"
	"log"
	"os"
	"strconv"
	"fmt"
	"time"
)

// Data is a custom struct
type Data struct {
	A int
	B float64
	C string
}

func main() {
	var n = 100000
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.Atoi(args[0])
	}

	var start = time.Now()

	for i := 0; i < n; i++ {
		var dataObj = Data{A: 13273828327, B: 382283.537749, C: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed laoreet luctus leo sed imperdiet. Morbi ut dolor eu arcu pretium bibendum. Donec eleifend arcu sit amet sodales ultrices. Nam quis diam vel mi hendrerit egestas quis in velit. Aliquam non vulputate magna. Cras et magna bibendum, facilisis magna et, rhoncus."}

		var buf bytes.Buffer
		var enc = gob.NewEncoder(&buf)
		var dec = gob.NewDecoder(&buf)
		// Encode
		var err = enc.Encode(dataObj)
		if err != nil {
			log.Fatal("encode error:", err)
		}
		// Decode
		var dataObj2 Data
		err = dec.Decode(&dataObj2)
		if err != nil {
			log.Fatal("decode error:", err)
		}
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/serialization/1.rs
use std::env;
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize, Debug)]
struct Data {
    a: i64,
    b: f64,
    c: String,
}

fn main() {
    let args: Vec<_> = env::args().collect();
    let n: usize = if args.len() > 1 {
        args[1].parse().unwrap()
    } else {
        100000
    };

    for _ in 0..n {
        let data = Data { a: 13273828327, b: 382283.537749, c: String::from("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed laoreet luctus leo sed imperdiet. Morbi ut dolor eu arcu pretium bibendum. Donec eleifend arcu sit amet sodales ultrices. Nam quis diam vel mi hendrerit egestas quis in velit. Aliquam non vulputate magna. Cras et magna bibendum, facilisis magna et, rhoncus.") };

        let encoded: Vec<u8> = bincode::serialize(&data).unwrap();

        #[cfg(debug_assertions)]
        println!("encoded = {:?}", encoded);

        let _decoded: Data = bincode::deserialize(&encoded[..]).unwrap();

        #[cfg(debug_assertions)]
        println!("decoded = {:?}", _decoded);
    }
}
